mongodbatlas/resource_mongodbatlas_privatelink_endpoint_serverless.go

```diff
@@ -0,0 +1,244 @@
+package mongodbatlas
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const (
+	errorServerlessEndpointAdd    = "error adding MongoDB Serverless PrivateLink Endpoint Connection(%s): %s"
+	errorServerlessEndpointDelete = "error deleting MongoDB Serverless PrivateLink Endpoint Connection(%s): %s"
+)
+
+func resourceMongoDBAtlasPrivateLinkEndpointServerless() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceMongoDBAtlasPrivateLinkEndpointServerlessCreate,
+		ReadContext:   resourceMongoDBAtlasPrivateLinkEndpointServerlessRead,
+		DeleteContext: resourceMongoDBAtlasPrivateLinkEndpointServerlessDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceMongoDBAtlasPrivateLinkEndpointServerlessImportState,
+		},
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"instance_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"endpoint_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"provider_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"AWS"}, false),
+			},
+			"endpoint_service_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(1 * time.Hour),
```

Comment:
Should we default timeouts to 2 hours and make configurable? similar as to what we did in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/859

Reply:
Fixed